### ISSUE TITLE
WebServiceTarget: Thread IDs do not match on Client and Server

### ISSUE THREAD

**Issue by MichelZ** [NLog/NLog#2111]
**-Bug**
**NLog version**:  4.4.9
**Platform**: .Net 3.5/4.0/4.5
**Current NLog config** (xml or C#, if relevant)

Client:
 ```
  var target = new LogReceiverWebServiceTarget
                       {
                           EndpointConfigurationName = "BasicHttpBinding_ILogReceiverServer",
                           EndpointAddress = webServiceEndpointAddress,
                           ClientId = null,
                           IncludeEventProperties = true,
                           Name = "WebServiceCore",
                           UseBinaryEncoding = true                           
                       };

            target.Parameters.Add(new MethodCallParameter("Client.date", "${date:universalTime=true:format=u}", typeof(string)));
            target.Parameters.Add(new MethodCallParameter("Client.processid", "${processid}", typeof(string)));
            target.Parameters.Add(new MethodCallParameter("Client.threadid", "${threadid}", typeof(string)));
            target.Parameters.Add(new MethodCallParameter("Client.level", "${level}", typeof(string)));
            target.Parameters.Add(new MethodCallParameter("Client.callsite", Branding.GetLoggingLayoutWithReplaceParams("${callsite:fileName=True:className=False:methodName=True:includeSourcePath=False:padding=50:padCharacter= }"), typeof(string)));
            target.Parameters.Add(new MethodCallParameter("Client.message", Branding.GetLoggingLayoutWithReplaceParams("${message}"), typeof(string)));
            target.Parameters.Add(new MethodCallParameter("Client.exception", "${exception:maxInnerExceptionLevel=3:format=Method,Message,StackTrace:innerFormat=Method,Message,StackTrace:separator=\r\n:innerExceptionSeparator=\r\n}", typeof(string)));
            target.Parameters.Add(new MethodCallParameter("Client.Module", "${gdc:Module}", typeof(string)));
            target.Parameters.Add(new MethodCallParameter("Client.MachineName", "${gdc:MachineName}", typeof(string)));

            return target;
```
Server:
```
return @"${event-context:item=Client.date:universalTime=true:format=u}|"
                   + "${event-context:item=Client.processid:fixedLength=True:padding=4:padCharacter= }|"
                   + "${event-context:item=Client.threadid:fixedLength=True:padding=3:padCharacter= }|"
                   + "${event-context:item=Client.level:fixedLength=True:padding=5:padCharacter= :upperCase=True}|"
                   + "${event-context:item=Client.callsite:fileName=True:className=False:methodName=True:includeSourcePath=False:padding=50:padCharacter= }|"
                   + "${event-context:item=Client.message}"
                   + "  ${event-context:item=Client.exception:maxInnerExceptionLevel=3:format=Method,Message,StackTrace:innerFormat=Method,Message," 
                   + "StackTrace:separator=\r\n:innerExceptionSeparator=\r\n}";
```

- What is the current result?
-> Thread ID's sometimes do not match between log on the client and log on the server
- What is the expected result?
-> Thread ID that is logged on the client gets sent to server and logged on the Server
- Did you checked the [Internal log](https://github.com/NLog/NLog/wiki/Internal-Logging)?
-> No
- Please post full exception details (message, stacktrace, inner exceptions)
-> No exceptions
- Are there any work arrounds?
-> No workarounds
- Is there a version in which it did worked? 
-> No
- Can you help us by writing an unit test?
-> I don't think this is "easily" unit-testable

**Comment by snakefoot** [comment]
Can see you add the target programmatically. Do you remember to call this method after having configured all targets:

`NLog.LogManager.ReconfigExistingLoggers();`

**Comment by snakefoot** [comment]
After looking at the NLog-code, then I think you can fix this by putting your LogReceiverWebServiceTarget inside a AsyncTargetWrapper.

Then it will correctly capture the current thread-context,

**Comment by MichelZ** [comment]
@snakefoot We are running ReconfigExistingLoggers(), yes.  We will try the AsyncTargetWrapper approach, thanks.

**Comment by 304NotModified** [comment]
Fixed in 4.4.10 and the release is queued. Will be online in a few hours!
